js/components/controlButton.js

```diff
@@ -0,0 +1,316 @@
+   * @private
+   * @return {Component} An AccessibleButton or NonAccessibleButton component, depending on the case
+   */
+  getButtonComponent: function() {
+    var Component;
+
+    if (this.props.ariaHidden === true) {
+      Component = NonAccessibleButton;
+    } else {
+      Component = AccessibleButton;
+    }
+    return Component;
+  },
+
+  /**
+   * Extracts the default styles for the button icon from the skin.json.
+   * @private
+   * @return {Object} An object with color and opacity properties which represent the default style of the button
+   */
+  getIconStyles: function() {
+    var iconStyles = {
+      color: Utils.getPropertyValue(this.props.skinConfig, 'controlBar.iconStyle.inactive.color'),
+      opacity: Utils.getPropertyValue(this.props.skinConfig, 'controlBar.iconStyle.inactive.opacity')
+    };
+    return iconStyles;
+  },
+
+  /**
+   * Applies the highlighted styles to the button icon on hover.
+   * TODO
+   * This was moved over from ControlBar but the original implementation sets the
+   * styles directly on the rendered element, which is an anti-pattern in React.
+   * Eventually we should update this in order to accomplish the same with setState().
+   * @private
+   * @param {Event} event The mousenter event object that triggered the action
+   */
+  highlight: function(event) {
+    if (
+      event.currentTarget.disabled ||
+      this.props.controller.state.isMobile
+    ) {
+      return;
+    }
+    var iconElement = Utils.getEventIconElement(event);
+
+    if (iconElement) {
+      var highlightOpacity = Utils.getPropertyValue(
+        this.props.skinConfig,
+        'controlBar.iconStyle.active.opacity',
+        1
+      );
+      var accentColor = Utils.getPropertyValue(
+        this.props.skinConfig,
+        'general.accentColor'
+      );
+      var highlightColor = Utils.getPropertyValue(
+        this.props.skinConfig,
+        'controlBar.iconStyle.active.color',
+        accentColor
+      );
+      Utils.highlight(iconElement, highlightOpacity, highlightColor);
+    }
+  },
+
+  /**
+   * Restores the default button styles after hover ends.
+   * @private
+   * @param {Event} event The mouseleave event object that triggered the action
+   */
+  removeHighlight: function(event) {
+    var iconElement = Utils.getEventIconElement(event);
+
+    if (iconElement) {
+      var baseOpacity = Utils.getPropertyValue(
+        this.props.skinConfig,
+        'controlBar.iconStyle.inactive.opacity'
+      );
+      var baseColor = Utils.getPropertyValue(
+        this.props.skinConfig,
+        'controlBar.iconStyle.inactive.color'
+      );
+      Utils.removeHighlight(iconElement, baseOpacity, baseColor);
+    }
+  },
+
+  /**
+   * Extracts the responsive UI multiple value from the skin config depending on the
+   * responsive view that is currently active.
+   * @private
+   * @return {Number} The numeric value of the UI multiple that matches the current responsive view.
```

Comment:
If "multiple" is referring to "multiplier," can we update the function name/jsdoc and variable names in the function to reference "multiplier" instead? Or if there is a decision to use "multiple," let me know.

Reply:
Yes I just copied this over from the old code. I think it should be multiplier. Will update this.